jsonrpc/dispatcher.go

```diff
@@ -202,11 +203,58 @@ func (d *Dispatcher) HandleWs(reqBody []byte, conn wsConn) ([]byte, error) {
 }
 
 func (d *Dispatcher) Handle(reqBody []byte) ([]byte, error) {
-	var req Request
-	if err := json.Unmarshal(reqBody, &req); err != nil {
+
+	x := bytes.TrimLeft(reqBody, " \t\r\n")
+	if len(x) == 0 {
 		return nil, invalidJSONRequest
 	}
-	return d.handleReq(req)
+	if x[0] == '{' {
+		var req Request
+		if err := json.Unmarshal(reqBody, &req); err != nil {
+			return nil, invalidJSONRequest
+		}
+		return d.handleReq(req)
+	}
+
+	// handle batch requests
+	var requests []Request
+	if err := json.Unmarshal(reqBody, &requests); err != nil {
+		return nil, d.internalError("batch method", err)
+	}
+	var responses []Response
+	for _, req := range requests {
+		var response, err = d.handleReq(req)
+		if err != nil {
+			d.logger.Debug("failed to dispatch", "method", "batch method", "err", err)
+			errorResponse := Response{
+				ID: req.ID,
+				JSONRPC: "2.0",
+				Error: internalError,
+			}
+			responses = append(responses, errorResponse)
+			continue
+		}
+	
+		// unmarshal response from handleReq so that we can re-marshal as batch responses
+		var resp Response
+		if err := json.Unmarshal(response, &resp); err != nil {
+			d.logger.Debug("failed to dispatch", "method", "batch method", "err", err)
```

Comment:
Call `internalError` instead?
```suggestion
			d.internalError("batch method", err)
```